Wildcard expansion with MSVC; thanks to Fucheng Leng (冷福成).
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -597,6 +597,32 @@
   target_sources(tvips2smv PRIVATE "getopt_long.c")
 endif()
 
+if(MSVC)
+  set_target_properties(
+    dumpframe
+    ht2wavelength
+    idoc2smv
+    img2px
+    mrc2smv
+    ser2smv
+    tiff2smv
+    tvips2smv
+    PROPERTIES
+    LINK_FLAGS "setargv.obj")
+endif()
+
+install(
+  TARGETS
+    dumpframe
+    ht2wavelength
+    idoc2smv
+    img2px
+    mrc2smv
+    ser2smv
+    tiff2smv
+    tvips2smv
+  RUNTIME)
+
 
 # To avoid automagic dependencies, Cython and its dependencies are
 # REQUIRED if the optional Python module is enabled.  There is no
@@ -894,17 +920,21 @@
   if(NOT HAVE_GETOPT_LONG)
     target_sources(tiff2raw PRIVATE "getopt_long.c")
   endif()
-endif()
-
 
-install(
-  TARGETS
-    dumpframe
-    ht2wavelength
-    idoc2smv
-    img2px
-    mrc2smv
-    ser2smv
-    tiff2smv
-    tvips2smv
-  RUNTIME)
+  if(MSVC)
+    set_target_properties(
+      cphdr
+      dan_test
+      fixpoint
+      img2img
+      img2splot
+      imgcorr
+      imgrevert
+      mike_test
+      rawdump
+      smv2pgm
+      tiff2raw
+      PROPERTIES
+      LINK_FLAGS "setargv.obj")
+  endif()
+endif()
